src/effects.c

```diff
@@ -2609,23 +2609,23 @@ static int ProcessGlobalClient(jack_nframes_t nframes, void *arg)
         memcpy(audio_out2_buf, audio_in2_buf, sizeof(float)*nframes);
         for (uint32_t i=0; i<nframes; ++i)
         {
-            gate_push_sample(&g_noisegate, audio_in1_buf[i], 0.f);
-            audio_out1_buf[i] = gate_run(&g_noisegate, audio_in1_buf[i]);
+            gate_push_samples(&g_noisegate, audio_in1_buf[i], 0.f);
```

Comment:
please add a `gate_push_sample` (singular/single) version of this call, so the useless comparison against 0 is skipped.

Reply:
actually a `gate_push_sample_and_apply` could work well here. since the "apply" step is just a multiplication return.